src/main/java/Duke.java

```diff
@@ -50,18 +54,29 @@ else if (input.equals("list")) {
                             if (input.equals("")) {
                                 throw new Missing("");
                             }
-                            input = input.replaceFirst( "/by", "(by:");
-                            input = input + ")";
+                            String[] taskDate = input.split("/by ");
+                            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("d/MM/yyyy HHmm");
+                            LocalDateTime dateTime = LocalDateTime.parse(taskDate[1], formatter);
+                            String dateTimeString = dateTime.format(DateTimeFormatter.ofPattern("MMM d yyyy hhmma"));
+                            input = taskDate[0] + "(by:" + dateTimeString  + ")";
                             newTask = new Deadline(input);
                             break;
                         case "event" :
```

Comment:
Just a minor coding standard violation but I think there should not be a white space between the "event" condition and the colon